pkg/discovery/repository/entity_metrics.go

```diff
@@ -82,10 +82,17 @@ func (namespaceMetrics *NamespaceMetrics) AggregateUsed(partialUsed map[metrics.
 // same ContainerSpec.
 type ContainerMetrics struct {
 	Capacity []float64
-	Used     []metrics.Point
+	// Used could be []metrics.Point or [][]metrics.ThrottlingCumulative
+	// [][]metrics. ThrottlingCumulative stores multipoints segregated per container.
+	// Storing it this way helps in identifying metrics per container while aggregating
+	// the metrics for containerSpecs.
+	// TODO: At some point this can be converted to a typed UsedMetric interface which
+	// implements the methods like len(), append() and getPoints() and have the implementation
+	// hidden from the users of this metric.
```

Comment:
+1
Perhaps we should consider doing it in part of this PR.  Is that a significant amount of work?  Leaving it as `interaface{}` seems a little hard to read.

Reply:
This change is relatively large; I actually started with this earlier, because I also think that a proper type is better in place of an abstract `interface{}` but settled with a TODO instead. To make it absolutely clean, we would need all the calculations (including whereever we use float64 for all our used metrics calculations) to be wrapped in to this `usedMetric` interface implementation. That means touching many places in code.

I will do this implementing but not as part of this PR as in an offline sync with @enlinxu, he pointed out to have this work moved ahead for test.

Reply:
Understood!